lib/CLIManager.js

```diff
@@ -777,6 +787,10 @@ class CLIManager {
       done(err);
     });
   }
+
+  processConfigFile(operationType, options, done) {
```

Comment:
I think that here you should either put the operationType inside the options object, or get everything out of the options object.

Also, if you decide to have the options object, I would change the ID of the object to always be "objectId", and not differ based on what type of configuration you are currently using it for (e.g. appId, envId, etc.)

Alternatively, you can have two separate methods. This is what I would do, as you do not seem to share any code between update and create, aside from the JSON parsing. Here are the methods:
processCreate(objectType, name, configFile);
processUpdate(objectType, objectId, configFile);

Reply:
I'd prefer keeping the `options` object and have it include the operationType.
As for the `objectId` - I'd rather not because I think that for different operations/configTypes I'd need different things and not neccesarily just one id.

Reply:
What different things do you expect to need for the different operations?

Reply:
In order to update the env the whole env object is needed. For creating dataLinks(services) we will probably need an appId or orgId and we will need to know whether it is an app or org id.
I feel that the file processor have to receive all the data it needs and to focus on processing the config files.

Reply:
The way I see it, we will not need any additional info. Aside from the name (which will later be the slug, once we have slugs), everything will be in the config file.
But OK, let's leave it like that for now. We'll see how it looks when we add another type of config file (service) and will change it if necessary.